arc-mlir/arc-mlir-build

```diff
@@ -41,6 +41,7 @@ A2M_ASSERTS="${A2M_ASSERTS:+-DLLVM_ENABLE_ASSERTIONS=ON}"
 
 # Do we want ccache?
 A2M_CCACHE="${A2M_CCACHE:+-DLLVM_CCACHE_BUILD=1}"
+A2M_CCACHE_MAXSIZE="${A2M_CCACHE:+-DLLVM_CCACHE_MAXSIZE=20G}"
```

Comment:
I'd prefer if this was only done on the CI-machine, users have their own preferences for the location and size of their ccache setup. If I did this I'd:

1. Figure out a path on the CI-machine which is writeable by the runner, but not zapped by it between runs and call this `<PERSIST_DIR>`

2. Create a shell script:
  ```
 #!/bin/bash
mkdir -p ${PERSIST_DIR}/ccache-cachedir
envsubst > ${PERSIST_DIR}/ccache-config <<EOF
max_size = 20G
cache_dir = ${PERSIST_DIR}/ccache-cachedir
EOF
  ```
3. In the configuration for the workflow environment I would set:
 ```
PERSIST_DIR="<PERSIST_DIR>"
CCACHE_CONFIGPATH="${PERSIST_DIR}/ccache-config"
```
And then configure the workflow to run the shell script before it runs `arc-build`